lightning/src/chain/channelmonitor.rs

```diff
@@ -695,6 +695,11 @@ pub(crate) struct ChannelMonitorImpl<Signer: Sign> {
 	// remote monitor out-of-order with regards to the block view.
 	holder_tx_signed: bool,
 
+	// If a spend of the funding output is seen, we set this to true and reject any further
+	// updates. This prevents any further changes in the offchain state no matter the order
+	// of block connection between ChannelMonitors and the ChannelManager.
+	funding_spend_seen: bool,
```

Comment:
There's a lot of state cleanup we can do in `ChannelMonitor`, which has largely been delayed just because its a delicate part of the code and easy to screw up. For now I kinda prefer the "simple, minimal code change" addition.